src/nvim/api/options.c

```diff
@@ -256,10 +256,10 @@ void nvim_set_option(uint64_t channel_id, String name, Object value, Error *err)
 /// @param name     Option name
 /// @param[out] err Error details, if any
 /// @return         Option value (global)
-Object nvim_get_option(String name, Error *err)
+Object nvim_get_option(uint64_t channel_id, String name, Error *err)
```

Comment:
for options you could remove the `uint64_t channel_id` argument and instead add `Arena* arena` argument and then use the non-allocating branch unconditionally (also for nvim_get_option_value)

Reply:
Is that so we avoid allocations on all targets?